sycl/test/xocc_tests/simple_tests/kernel_arguement_forwarding.cpp

```diff
@@ -0,0 +1,37 @@
+// REQUIRES: xocc
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out 2>&1 | FileCheck %s
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/xilinx/fpga.hpp>
+#include <array>
+#include <iostream>
+#include <tuple>
+#include <utility>
+
+using namespace sycl::xilinx::literal;
+
+int main() {
+  sycl::buffer<sycl::cl_int, 1> Buffer(4);
+  sycl::queue Queue;
+  sycl::range<1> NumOfWorkItems{Buffer.get_count()};
+
+  Queue.submit([&](sycl::handler &cgh) {
+    auto Accessor = Buffer.get_access<sycl::access_mode::write>(cgh);
+    cgh.single_task<class FirstKernel>([=]() {
+      "--optimize 2"_kernel_param;
+      // CHECK: v++ {{.*}}class_FirstKernel{{.*}} --optimize 2
+      Accessor[0] = 0;
+    });
+  });
+
+  Queue.submit([&](sycl::handler &cgh) {
+    auto Accessor = Buffer.get_access<sycl::access_mode::write>(cgh);
+    cgh.single_task<class SecondKernel>([=]() {
```

Comment:
As an alternative syntax, it would be nice to have also a decorator syntax to the kernel, etc.